Core/src/IfcGeometryConverter/GeomUtils.cpp

```diff
@@ -659,48 +668,76 @@ bool LineToLineIntersectionHelper(carve::geom::vector<2>& v1,
+												const carve::geom::vector<2>& initialPointSeg2,
+												const carve::geom::vector<2>& terminalPointSeg2,
+	carve::geom::vector<2> & intersectionPoint)
 {
-	double r, s;
-	if( LineToLineIntersectionHelper(v1, v2, v3, v4, r, s) )
+	double distToIntesection1, distToIntesection2;
+	if( LineToLineIntersectionHelper(initialPointSeg1, terminalPointSeg1, initialPointSeg2, terminalPointSeg2, distToIntesection1, distToIntesection2) )
 	{
-		if (r >= 0 && r <= 1)
+		if (distToIntesection1 >= 0 && distToIntesection1 <= 1)
 		{
-			result.push_back(v1 + (v2 - v1) * r);
+			intersectionPoint = (initialPointSeg1 + (terminalPointSeg1 - initialPointSeg1) * distToIntesection1);
 			return true;
 		}
 	}
 	return false;
 }
 
 /**********************************************************************************************/
-
-bool GeomUtils::LineSegmentToLineSegmentIntersection(carve::geom::vector<2>& v1,
-													 carve::geom::vector<2>& v2,
-													 carve::geom::vector<2>& v3, 
-													 carve::geom::vector<2>& v4, 
-										std::vector<carve::geom::vector<2> >& result )
+/*! \brief Calculates coordinates of the intersection point.
+* \param[in] initialPointSeg1		Initial point of the first intersecting segment.
+* \param[in] terminalPointSeg1		Terminal point of the first intersecting segment.
+* \param[in] initialPointSeg2		Initial point of the second intersecting segment.
+* \param[in] terminalPointSeg2		Terminal point of the second intersecting segment.
+* \param[out] intersectionPoint		Coordinates of the intersection point between first line segment and second line segment.
+*/
+bool GeomUtils::LineSegmentToLineSegmentIntersection(const carve::geom::vector<2>& initialPointSeg1,
+														const carve::geom::vector<2>& terminalPointSeg1,
+														const carve::geom::vector<2>& initialPointSeg2,
+														const carve::geom::vector<2>& terminalPointSeg2,
+	carve::geom::vector<2> & intersectionPoint)
 {
-	double r, s;
-	if( LineToLineIntersectionHelper(v1, v2, v3, v4, r, s) )
+	double distToIntesection1, distToIntesection2;
+	if( LineToLineIntersectionHelper(initialPointSeg1, terminalPointSeg1, initialPointSeg2, terminalPointSeg2, distToIntesection1, distToIntesection2) )
 	{
-		if (r >= 0 && r <= 1)
+		if (distToIntesection1 >= 0 && distToIntesection1 <= 1)
 		{
-			if (s >= 0 && s <= 1)
+			if (distToIntesection2 >= 0 && distToIntesection2 <= 1)
 			{
-				result.push_back(v1 + (v2 - v1) * r);
+				intersectionPoint = (initialPointSeg1 + (terminalPointSeg1 - initialPointSeg1) * distToIntesection1);
 				return true;
 			}
 		}
 	}
 	return false;
 }
 
+bool GeomUtils::LineToLineIntersection(const carve::geom::vector<2>& initialPointSeg1,
+										const carve::geom::vector<2>& terminalPointSeg1,
+										const carve::geom::vector<2>& initialPointSeg2,
+										const carve::geom::vector<2>& terminalPointSeg2,
+										carve::geom::vector<2> & intersectionPoint)

```

Comment:
- doxygen comment missing
- please make sure, that the indentation is consistent

Reply:
Resolved:  a619817